File: flutter_app/lib/models/acte_medical.dart
class ActeMedical {
  final int id;
  final int sejourId;
  final String code;
  final String libelle;
  final String type;
  final String dateRealisation;
  final double tarif;
  final String? medecin;
  final String? notes;

  ActeMedical({
    required this.id,
    required this.sejourId,
    required this.code,
    required this.libelle,
    required this.type,
    required this.dateRealisation,
    required this.tarif,
    this.medecin,
    this.notes,
  });

  factory ActeMedical.fromJson(Map<String, dynamic> json) {
    return ActeMedical(
      id: json['id'] ?? 0,
      sejourId: json['sejourId'] ?? 0,
      code: json['code'] ?? '',
      libelle: json['libelle'] ?? '',
      type: json['type'] ?? '',
      dateRealisation: json['dateRealisation'] ?? '',
      tarif: (json['tarif'] ?? 0).toDouble(),
      medecin: json['medecin'],
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'sejourId': sejourId,
      'code': code,
      'libelle': libelle,
      'type': type,
      'dateRealisation': dateRealisation,
      'tarif': tarif,
      'medecin': medecin,
      'notes': notes,
    };
  }

  ActeMedical copyWith({
    int? id,
    int? sejourId,
    String? code,
    String? libelle,
    String? type,
    String? dateRealisation,
    double? tarif,
    String? medecin,
    String? notes,
  }) {
    return ActeMedical(
      id: id ?? this.id,
      sejourId: sejourId ?? this.sejourId,
      code: code ?? this.code,
      libelle: libelle ?? this.libelle,
      type: type ?? this.type,
      dateRealisation: dateRealisation ?? this.dateRealisation,
      tarif: tarif ?? this.tarif,
      medecin: medecin ?? this.medecin,
      notes: notes ?? this.notes,
    );
  }

  @override
  String toString() => 'ActeMedical(id: $id, code: $code, libelle: $libelle)';
}

File: backend/src/main/java/com/healthcare/dashboard/entities/ActeMedical.java
package com.healthcare.dashboard.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "actes_medicaux")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ActeMedical {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "sejour_id", nullable = false)
    private Sejour sejour;
    
    @Column(nullable = false)
    private String code;
    
    @Column(nullable = false)
    private String libelle;
    
    @Column(nullable = false)
    private String type; // CONSULTATION, CHIRURGIE, RADIOLOGIE, LABORATOIRE, etc.
    
    @Column(nullable = false)
    private LocalDateTime dateRealisation;
    
    @Column(nullable = false)
    private Double tarif;
    
    private String medecin;
    
    private String notes;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
